lib/core/src/sdk.rs

```diff
@@ -1140,32 +1277,63 @@ impl LiquidSdk {
+        req: &PrepareReceiveRequest,
+    ) -> Result<PrepareReceiveResponse, PaymentError> {
         self.ensure_is_started().await?;
-        let reverse_pair = self
-            .swapper
-            .get_reverse_swap_pairs()?
-            .ok_or(PaymentError::PairsNotFound)?;
 
-        let payer_amount_sat = req.payer_amount_sat;
-        let fees_sat = reverse_pair.fees.total(req.payer_amount_sat);
+        let fees_sat;
+        let amount_sat = req.amount_sat;
+        match req.payment_method {
+            PaymentMethod::Lightning => {
+                let Some(amount_sat) = req.amount_sat else {
+                    return Err(PaymentError::AmountMissing { err: "`amount_sat` must be specified when `PaymentMethod::Lightning` is used.".to_string() });
+                };
+                let reverse_pair = self
+                    .swapper
+                    .get_reverse_swap_pairs()?
+                    .ok_or(PaymentError::PairsNotFound)?;
+
+                fees_sat = reverse_pair.fees.total(amount_sat);
 
-        ensure_sdk!(payer_amount_sat > fees_sat, PaymentError::AmountOutOfRange);
+                ensure_sdk!(amount_sat > fees_sat, PaymentError::AmountOutOfRange);
 
-        reverse_pair
-            .limits
-            .within(payer_amount_sat)
-            .map_err(|_| PaymentError::AmountOutOfRange)?;
+                reverse_pair
+                    .limits
+                    .within(amount_sat)
+                    .map_err(|_| PaymentError::AmountOutOfRange)?;
 
-        debug!("Preparing Receive Swap with: payer_amount_sat {payer_amount_sat} sat, fees_sat {fees_sat} sat");
+                debug!(
+                    "Preparing Lightning Receive Swap with: amount_sat {amount_sat} sat, fees_sat {fees_sat} sat"
+                );
+            }
+            PaymentMethod::BitcoinAddress => {
+                let Some(amount_sat) = req.amount_sat else {
+                    return Err(PaymentError::AmountMissing { err: "`amount_sat` must be specified when `PaymentMethod::BitcoinAddress` is used.".to_string() });
+                };
+                let pair = self.validate_chain_pairs(Direction::Incoming, amount_sat)?;
+                let claim_fees_sat = pair.fees.claim_estimate();
+                let server_fees_sat = pair.fees.server();
+                fees_sat = pair.fees.boltz(amount_sat) + claim_fees_sat + server_fees_sat;
+                debug!(
+                    "Preparing Chain Receive Swap with: amount_sat {amount_sat} sat, fees_sat {fees_sat} sat"
+                );
+            }
+            PaymentMethod::LiquidAddress => {
+                fees_sat = 0;
+                debug!(
+                    "Preparing Liquid Receive Swap with: amount_sat {amount_sat:?} sat, fees_sat {fees_sat} sat"
+                );
+            }
+        };
 
-        Ok(PrepareReceivePaymentResponse {
-            payer_amount_sat,
+        Ok(PrepareReceiveResponse {
+            amount_sat,
```

Comment:
Why was this renamed from `payer_amount_sat`? AFAIK it still represents the payer amount.

Reply:
Indeed, might get confused with the receiver amount. I'll rollback the change.